### ISSUE TITLE
False unsubscriptable error on types

### ISSUE THREAD

**Issue by gnvk** [PyCQA/pylint#3240]
### Steps to reproduce
``` python
from typing import Any
import subprocess

def run(args: Any) -> subprocess.CompletedProcess[bytes]:
    return subprocess.run(args, check=True)
```

### Current behavior
E1136: Value 'subprocess.CompletedProcess' is unsubscriptable (unsubscriptable-object)

### Expected behavior
No errors, the type CompletedProcess is generic

### pylint --version output
pylint 2.4.3
astroid 2.3.3
Python 3.6.5 (default, Aug 27 2018, 16:16:07)
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]

**Comment by PCManticore** [comment]
@gnvk I'm not sure what you mean by `CompletedProcess` being a generic class. It is not and it does not seem subscriptable: https://github.com/python/cpython/blob/3.8/Lib/subprocess.py#L415

Running the code will also trigger a `TypeError`:
```
Traceback (most recent call last):
  File "a.py", line 4, in <module>
    def run(args: Any) -> subprocess.CompletedProcess[bytes]:
TypeError: 'type' object is not subscriptable
```

While annotations might not be required to be Python code, `pylint` takes a different stances and tries to make sure that even annotations will not result in runtime errors.
